modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/ImagePins.js

```diff
@@ -100,48 +136,231 @@ const ImagePins = ({
 			zoomIn(containerRef.current, panZoomRef.current);
 		}
 
-		handlers.current = {
-			moveController: (where) =>
-				moveController(
-					containerRef.current,
-					navigationController,
-					where
-				),
-			zoomIn: () => zoomIn(containerRef.current, panZoomRef.current),
-			zoomOut: () => zoomOut(containerRef.current, panZoomRef.current),
+		const clickAction = (updatedPin) =>
+			setShowTooltip({
+				details: {
+					cx: updatedPin.cx,
+					cy: updatedPin.cy,
+					id: updatedPin.id,
+					label: updatedPin.label,
+					linked_to_sku: updatedPin.linked_to_sku,
+					quantity: updatedPin.quantity,
+					sku: updatedPin.sku,
+				},
+				tooltip: true,
+			});
+
+		function dragstarted() {
+			select(this).raise().classed('active', true);
+		}
+
+		function dragged() {
+			select(this).attr(
+				'transform',
+				'translate(' + event.x + ',' + event.y + ')'
+			);
+		}
+
+		function dragended() {
```

Comment:
usually we use lower camel case for methods, `dragEnded`

Reply:
can i refactor this in the 3of5 pr if it's not blocking? we are trying to go forward to implement this in a lr module

Reply:
Yep, go ahead, nothing I suggested is blocking.